qiskit/quantum_info/operators/symplectic/pauli_list.py

```diff
@@ -128,16 +128,20 @@ def __init__(self, data):
             can share the same underlying array.
         """
         if isinstance(data, BasePauli):
+            if num_qubits is not None and num_qubits != data.num_qubits:
+                raise ValueError("Provided num_qubits does not match the input.")
```

Comment:
Generally I like to see the failing values etc formatted into the message to give that extra info which may be helpful to the user in understanding/sorting the failure out.